src/overlays/actors/ovl_Boss_Mo/z_boss_mo.c

```diff
@@ -1,15 +1,121 @@
+/*
+ * File: z_boss_mo.c
+ * Overlay: ovl_Boss_Mo
+ * Description: Morpha
+ */
+
 #include "z_boss_mo.h"
+#include "objects/object_mo/object_mo.h"
+#include "objects/gameplay_keep/gameplay_keep.h"
+#include "vt.h"
 
 #define FLAGS 0x00000035
 
 #define THIS ((BossMo*)thisx)
 
+#define MO_WATER_LEVEL(globalCtx) globalCtx->colCtx.colHeader->waterBoxes[0].ySurface
+
+#define HAS_LINK(tent) ((tent != NULL) && ((tent->actionState == MO_TENT_GRAB) || (tent->actionState == MO_TENT_SHAKE)))
+
+typedef struct BossMoEffect {
+    /* 0x00 */ Vec3f pos;
+    /* 0x0C */ Vec3f vel;
+    /* 0x18 */ Vec3f accel;
+    /* 0x24 */ u8 type;
+    /* 0x25 */ u8 timer;
+    /* 0x26 */ u8 stopTimer;
+    /* 0x28 */ s16 unk_28; // unused?
+    /* 0x2A */ s16 alpha;
+    /* 0x2C */ s16 rippleMode;
+    /* 0x2E */ s16 maxAlpha;
+    /* 0x30 */ f32 scale;
+    /* 0x34 */ f32 moFxFloat1;
+    /* 0x38 */ f32 moFxFloat2;
```

Comment:
Wouldn't it make more to define these as an array, at least to avoid the few `(*effect).` that are required in the file?